packages/contracts-bedrock/test/L2/GasPriceOracle.t.sol

```diff
@@ -93,3 +102,111 @@ contract GasPriceOracle_Test is CommonTest {
         assertEq(returndata, hex"");
     }
 }
+
+contract GasPriceOracleEcotone_Test is GasPriceOracle_Test {
+    /// @dev Sets up the test suite.
+    function setUp() public virtual override {
+        super.setUp();
+
+        // Define the function signature
+        bytes4 functionSignature = bytes4(keccak256("setL1BlockValuesEcotone()"));
+
+        // Encode the function signature and extra data
+        bytes memory callDataPacked = abi.encodePacked(
+            basefeeScalar,
```

Comment:
A nice to have would be a solidity library function that encodes the data as expected and use that everywhere instead of the 2 calls to encodePacked. The function could exist in `libraries/Encoding.sol` and be called `encodeSetL1BlockValuesEcotone` and return `bytes memory`

Reply:
If we had such a library, it would make differentially fuzzing against a Go implementation really easy to do

Reply:
two calls to encodePacked weren't actually needed, I have simplified it to one and moved it into libaries/Encoding.sol.